protocol/x/clob/keeper/liquidations.go

```diff
@@ -106,28 +106,28 @@ func (k Keeper) LiquidateSubaccountsAgainstOrderbook(
 		}
 
 		if optimisticallyFilledQuantums == 0 {
-			if numAttemptedDeleveraging < k.Flags.MaxDeleveragingAttemptsPerBlock {
-				// The liquidation order was unfilled. Try to deleverage the subaccount.
-				deltaQuantums := liquidationOrder.GetDeltaQuantums()
-
-				// Get a pseudorandom perpetualId from the list of open positions.
-				subaccount := k.subaccountsKeeper.GetSubaccount(ctx, subaccountId)
-				positions := subaccount.GetPerpetualPositions()
-				perpetualId := positions[pseudoRand.Intn(len(positions))].PerpetualId
-
-				_, err := k.MaybeDeleverageSubaccount(ctx, subaccountId, perpetualId, deltaQuantums)
-				if err != nil {
-					k.Logger(ctx).Error(
-						"Failed to deleverage subaccount.",
-						"subaccount", subaccountId,
-						"perpetualId", perpetualId,
-						"deltaQuantums", deltaQuantums,
-						"error", err,
-					)
-					return err
-				}
-				numAttemptedDeleveraging++
-			}
+			unfilledLiquidations = append(unfilledLiquidations, *liquidationOrder)
+		}
+	}
+
+	// For each unfilled liquidation, attempt to deleverage the subaccount.
+	for i := 0; i < int(k.Flags.MaxDeleveragingAttemptsPerBlock) && i < len(unfilledLiquidations); i++ {
+		liquidationOrder := unfilledLiquidations[i]
+
+		subaccountId := liquidationOrder.GetSubaccountId()
+		perpetualId := liquidationOrder.MustGetLiquidatedPerpetualId()
+		deltaQuantums := liquidationOrder.GetDeltaQuantums()
+
+		_, err := k.MaybeDeleverageSubaccount(ctx, subaccountId, perpetualId, deltaQuantums)
```

Comment:
btw one thing in 0.3.X I did was use the entire position size as `deltaQuantums` to deleverage as fast as possible. This is preferable to using the liquidation order because 1. we already have a limit on the number of accounts iterated for deleveraging and 2. there is no orderbook impact to really worry about

Reply:
makes sense. Updated!